test_many_concurrent_qstat_invocations is flaky

**Describe the bug**
A test is flaky, at least when run together with `pytest-xdist`

`tests/unit_tests/c_wrappers/res/job_queue/test_ert_qstat_proxy.py::test_many_concurrent_qstat_invocations[834-1000]
`

**To reproduce**
Steps to reproduce the behaviour:

`pytest -x -n 24 tests/unit_tests/c_wrappers/res/job_queue/test_ert_qstat_proxy.py  --count 1000`

Gave after a few minutes (on a beefy ubuntu 22.04 workstation):
    
 ```
               else:
                    assert state == CacheState.CACHE_EXISTS
>                   assert (
                        process.returncode == 0
                    ), "Check for race condition if AssertionError"
E                   AssertionError: Check for race condition if AssertionError
E                   assert 1 == 0
E                    +  where 1 = <Popen: returncode: 1 args: [PosixPath('/home/berland/projects/ert/src/ert/_...>.returncode

/home/berland/projects/ert/tests/unit_tests/c_wrappers/res/job_queue/test_ert_qstat_proxy.py:311: AssertionError
--------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------
011111111111111111111111111111111111111110

```


**Expected behaviour**
No error

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment**
 - OS: Ubuntu 22.04
 - ERT/Komodo release: main
 - Python version: 3.10
 - Remote/HPC execution involved: no

**Additional context**
Add any other context about the problem here.

--- a/tests/unit_tests/c_wrappers/res/job_queue/test_ert_qstat_proxy.py
+++ b/tests/unit_tests/c_wrappers/res/job_queue/test_ert_qstat_proxy.py
@@ -259,7 +259,7 @@ def test_many_concurrent_qstat_invocations(tmpdir, monkeypatch):
     """
     starttime = time.time()
     invocations = 400
-    sleeptime = 0.01  # seconds. Lower number increase probability of race conditions.
+    sleeptime = 0.02  # seconds. Lower number increase probability of race conditions.
     # (the mocked qstat backend sleeps for 0.5 seconds to facilitate races)
     cache_timeout = 2  # This is CACHE_TIMEOUT in the shell script
     assert invocations * sleeptime > cache_timeout  # Ensure race conditions can happen
@@ -277,6 +277,8 @@ def test_many_concurrent_qstat_invocations(tmpdir, monkeypatch):
                     stdout=subprocess.DEVNULL,
                 )
             )
+            # The asserts below in this test function depend on each subprocess
+            # to finish in order of invocation:
             time.sleep(sleeptime)
 
         class CacheState(enum.Enum):

